Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/SendSearchRequestToDrivers/Handle/Internal/DriverPool.hs

```diff
@@ -83,56 +87,69 @@ prepareDriverPoolBatch driverPoolCfg searchReq searchTry batchNum = withLogTag (
   logDebug $ "PreviousBatchesDrivers-" <> show previousBatchesDrivers
   prepareDriverPoolBatch' previousBatchesDrivers
   where
+    getPreviousBatchesDrivers = do
+      batches <- previouslyAttemptedDrivers searchTry.id
+      return $ (.driverPoolResult.driverId) <$> batches
+
     prepareDriverPoolBatch' previousBatchesDrivers = do
       radiusStep <- getPoolRadiusStep searchReq.id
       transporterConfig <- TC.findByMerchantId searchReq.providerId >>= fromMaybeM (TransporterConfigDoesNotExist searchReq.providerId.getId)
       intelligentPoolConfig <- DIP.findByMerchantId searchReq.providerId >>= fromMaybeM (InternalError "Intelligent Pool Config not found")
-
       blockListedDrivers <- Redis.withCrossAppRedis $ Redis.getList (mkBlockListedDriversKey searchReq.id)
       logDebug $ "Blocked Driver List-" <> show blockListedDrivers
-
-      allNearbyDriversCurrentlyNotOnRide <- calcDriverPool radiusStep
-      let reduceRadiusValue = driverPoolCfg.radiusShrinkValueForDriversOnRide
-      allNearbyDriversCurrentlyOnRide <- calcDriverCurrentlyOnRidePool radiusStep reduceRadiusValue transporterConfig
-      let driverPoolList = allNearbyDriversCurrentlyOnRide ++ allNearbyDriversCurrentlyNotOnRide
-      let allNearbyDrivers = filter (\dpr -> dpr.driverPoolResult.driverId `notElem` blockListedDrivers) driverPoolList
-      let sortingType = driverPoolCfg.poolSortingType
-      let batchSize = driverPoolCfg.driverBatchSize
-      logDebug $ "DriverPool-" <> show allNearbyDrivers
-      let onlyNewDrivers = filter (\dpr -> dpr.driverPoolResult.driverId `notElem` previousBatchesDrivers) allNearbyDrivers
-      driverPoolWithoutBatchSplitDelays <-
-        if length onlyNewDrivers < batchSize
-          then do
-            isAtMaxRadiusStep' <- isAtMaxRadiusStep radiusStep
-            if isAtMaxRadiusStep'
-              then calculatePool batchSize sortingType onlyNewDrivers allNearbyDrivers intelligentPoolConfig transporterConfig
-              else do
-                incrementPoolRadiusStep searchReq.id
-                prepareDriverPoolBatch' previousBatchesDrivers
-          else calculatePool batchSize sortingType onlyNewDrivers allNearbyDrivers intelligentPoolConfig transporterConfig
-      pure $ addDistanceSplitConfigBasedDelaysForDriversWithinBatch driverPoolWithoutBatchSplitDelays
+      goHomeConfig <- CQGHC.findByMerchantId searchReq.providerId
+      allNearbyGoHomeDrivers <-
```

Comment:
lets do nearby drivers query here and pass the list as argument to the below pool calculation functions -> `calcGoHomeDriverPool`, `calcDriverPool` and `calcDriverCurrentlyOnRidePool`

Reply:
As we discussed, GoTo Ride radius is different from normal radius so this cannot be implemented.